pkg/providers/k8s/k8s.go

```diff
@@ -0,0 +1,147 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/helmfile/vals/pkg/api"
+	"github.com/helmfile/vals/pkg/log"
+)
+
+type provider struct {
+	log            *log.Logger
+	KubeConfigPath string
+	KubeContext    string
+}
+
+func New(l *log.Logger, cfg api.StaticConfig) *provider {
+	p := &provider{
+		log: l,
+	}
+
+	kubeConfig, err := getKubeConfig(cfg)
+	if err != nil {
+		fmt.Printf("An error occurred getting the Kubeconfig path: %s\n", err)
+		return p
+	}
+
+	p.KubeConfigPath = kubeConfig
+	p.KubeContext = getKubeContext(cfg)
+
+	return p
+}
+
+func getKubeConfig(cfg api.StaticConfig) (string, error) {
+	// Use kubeConfigPath from URI parameters if specified
+	if cfg.String("kubeConfigPath") != "" {
+		if _, err := os.Stat(cfg.String("kubeConfigPath")); err != nil {
+			return cfg.String("kubeConfigPath"), fmt.Errorf("kubeConfigPath URI parameter is set but path %s does not exist.", cfg.String("kubeConfigPath"))
+		}
+	}
+
+	// Use path in KUBECONFIG environment variable if set
+	if envPath := os.Getenv("KUBECONFIG"); envPath != "" {
```

Comment:
The kubectl KUBECONFIG env variable impl supports a list of paths
https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable
I think we should support the same pattern here

Reply:
I can create an issue for it and see to picking it up at a later time if that's alright.